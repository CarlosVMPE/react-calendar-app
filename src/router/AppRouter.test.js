import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { CalendarPage } from '../calendar';
import { useAuthStore } from "../hooks/useAuthStore";
import { AppRouter } from "./AppRouter";

jest.mock('../hooks/useAuthStore');
//* MOCK del Componente
jest.mock('../calendar', () => ({
    CalendarPage: () => <h1>CalendarPage</h1>
}));

describe('Testing in AppRouter', () => {

    const mockCheckAuthToken = jest.fn();

    beforeEach(() => jest.clearAllMocks());

    test('should be show the loading page and call checkAuthToken', () => {
        useAuthStore.mockReturnValue({
            status: 'checking',
            checkAuthToken: mockCheckAuthToken
        })

        render(<AppRouter />);
        //screen.debug();

        expect(screen.getByText('Cargando...')).toBeTruthy();
        expect(mockCheckAuthToken).toHaveBeenCalled();
    });

    test('should be show the loading page and call checkAuthToken', () => {
        useAuthStore.mockReturnValue({
            status: 'not-authenticated',
            checkAuthToken: mockCheckAuthToken
        })

        const { container } = render(
            <MemoryRouter initialEntries={['/auth2/algo/otracosa']}>
                <AppRouter />
            </MemoryRouter>
        );
        //screen.debug();
        expect(screen.getByText('Ingreso')).toBeTruthy();
        expect(container).toMatchSnapshot();
    });

    test('should be show the calendar page if we are authenticated', () => {
        useAuthStore.mockReturnValue({
            status: 'authenticated',
            checkAuthToken: mockCheckAuthToken
        })

        render(
            <MemoryRouter>
                <AppRouter />
            </MemoryRouter>
        );
        expect(screen.getByText('CalendarPage')).toBeTruthy();
    });


})