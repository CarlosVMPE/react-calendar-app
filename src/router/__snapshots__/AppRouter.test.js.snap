// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in AppRouter should be show the loading page and call checkAuthToken 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 login-form-1"
      >
        <h3>
          Ingreso
        </h3>
        <form
          aria-label="login-form"
        >
          <div
            class="form-group mb-2"
          >
            <input
              aria-label="loginEmail"
              class="form-control"
              name="loginEmail"
              placeholder="Correo"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              data-testid="loginPassword"
              name="loginPassword"
              placeholder="Contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-3 text-center"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        class="col-md-6 login-form-2"
      >
        <h3>
          Registro
        </h3>
        <form
          aria-label="register-form"
        >
          <div
            class="form-group mb-2"
          >
            <input
              aria-label="registerName"
              class="form-control"
              name="registerName"
              placeholder="Nombre"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              aria-label="registerEmail"
              class="form-control"
              name="registerEmail"
              placeholder="Correo"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              data-testid="registerPassword"
              name="registerPassword"
              placeholder="Contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              data-testid="registerPassword2"
              name="registerPassword2"
              placeholder="Repita la contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-3 text-center"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Crear cuenta"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
